src/js/core/player.js

```diff
@@ -792,17 +792,53 @@ function name(): {country: string, firstName: string, lastName: string} {
     const fnRand = random.uniform(0, playerNames.first[country][playerNames.first[country].length - 1][1]);
     const firstNameRow = playerNames.first[country].find(row => row[1] >= fnRand);
     if (firstNameRow === undefined) {
-        throw new Error(`Undefined firstNameRow (fnRand=${fnRand}`);
+        throw new Error(`Undefined firstNameRow (fnRand=${fnRand})`);
+    }
+    let firstName = firstNameRow[0];
+
+    // Middle Name, randomly
+    const middle = random.uniform(0, 1);
+    if (middle < 0.05) {
+        const mnRand = random.uniform(0, playerNames.first[country][playerNames.first[country].length - 1][1]);
+        const middleNameRow = playerNames.first[country].find(row => row[1] >= mnRand);
+        if (middleNameRow === undefined) {
+            throw new Error(`Undefined middleNameRow (mnRand=${mnRand})`);
+        }
+        let middleName = middleNameRow[0];
+
+        // abbreviation rules:
+        // 50% will initialize their middle name
+        //   e.g. {Xavier Y. Zed}
+        // 50% will use it fully
+        //   e.g. {Xavier Yanick Zed}
+        const abbrev = random.uniform(0, 1);
+        if (abbrev < 0.5 || country === "USA") {
+            // initialize it
+            middleName = `${middleName.charAt(0)}.`;
+        } else {
+            // use it fully
+        }
+        firstName = `${firstName} ${middleName}`;
     }
-    const firstName = firstNameRow[0];
 
     // Last name
     const lnRand = random.uniform(0, playerNames.last[country][playerNames.last[country].length - 1][1]);
     const lastNameRow = playerNames.last[country].find(row => row[1] >= lnRand);
     if (lastNameRow === undefined) {
-        throw new Error(`Undefined lastNameRow (lnRand=${lnRand}`);
+        throw new Error(`Undefined lastNameRow (lnRand=${lnRand})`);
+    }
+    let lastName = lastNameRow[0];
+
+    // Jr. or III, randomly
+    const son = random.uniform(0, 1);
+    const english = ["USA", "Canada", "Australia", "New Zealand", "England", "Ireland", "Jamaica", "Phillippines"];
+    if (english.indexOf(country) > 0 && son < 0.01) {
+        if (son < 0.004) {
+            lastName = `${lastName} III`;
```

Comment:
Working on this addition now. Not a perfect source but I found this passage:

> "An informal poll by Joe Hyams, Jr., in 1958, disclosed that about fifty percent of the Juniors of famous or successful fathers were generally opposed to the practice, **about twenty-five percent were in favor**, and the remaining twenty-five percent saw both good and bad features in the situation."

https://books.google.com/books?id=OfHBWBK1wQQC&pg=PA279

I'll use 25% in a geometric progression. So suppose the base value of `son` is 1%, then random values between 0.0025 and 0.01 (~0.75% of all players, 1 in 133) will have the base level of Jr, then between .000625 and .0025 (~0.18% of all players, 1 in 555) will have the next level of III.

Then 1 in 2220 are IV, 1 in 8880 are V, and so on. Repeating this until John Smith XI is less than 1 in a billion players.

Should be enough? I considered going up to one trillion, which is around John Smith XVII.